odl/deform/linearized_deform.py

```diff
@@ -0,0 +1,497 @@
+        assert isinstance(displacement.space, odl.ProductSpace)
+
+        domain_space = displacement[0].space
+        range_space = domain_space
+
+        Operator.__init__(self, domain_space, range_space, linear=True)
+
+        self.displacement = displacement
+
+    def _call(self, template):
+        """Implementation of ``self(template)``.
+
+        Parameters
+        ----------
+        template : `DiscreteLpVector`
+            Given template that is to be deformed by the fixed
+            displacement field.
+        """
+        return linear_deform(template, self.displacement)
+
+    @property
+    def adjoint(self):
+        """Adjoint of the operator.
+
+        Returns
+        -------
+        adj_op : `Operator`
+            The adjoint of the operator.
+        """
+        adj_op = LinDeforFixedDispAdj(self.displacement)
+        return adj_op
+
+
+class LinDeforFixedDispAdj(Operator):
+    """Adjoint of the fixed displacement linearized deformation operator.
+
+    This operator computes the adjoint of the linear operator that
+    map a template ``I`` to its deformation using a fixed displacement ``v``:
+
+        I --> det(D(Id - v)) * I(Id - v)
+
+    Here, the ``I`` is the given template, ``Id`` is the identity mapping:
+
+        y --> y
+
+    the vector field ``v`` is a displacement field:
+
+        y --> v(y)
+
+    and, ``det(D(Id - v)) = 1 - div(v)`` is the determinant of the
+    Jacobian of ``Id - v``. Note that ``(Id + v)^{-1}`` is approximated
+    by ``(Id - v)``, which is accurate in some sense if the magnitude
+    of``v`` is close to ``0`` .
+    """
+    def __init__(self, displacement):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        displacement : `ProductSpaceVector`
+            Fixed displacement field used in the linearized deformation.
+
+        Examples
+        --------
+        >>> import odl
+        >>> space = odl.uniform_discr(0, 1, 5)
+        >>> disp_field_space = odl.ProductSpace(space, space.ndim)
+        >>> displacement_field = disp_field_space.element([[0, 0, 0, -0.2, 0]])
+        >>> op = LinDeforFixedDispAdj(displacement_field)
+        >>> template = space.element([0, 0, 1, 0, 0])
+        >>> op(template)
+        uniform_discr(0.0, 1.0, 5).element([0.0, 0.0, 2.0, 0.0, 0.0])
+        """
+        assert isinstance(displacement.space, odl.ProductSpace)
+
+        domain_space = displacement[0].space
+        range_space = domain_space
+
+        Operator.__init__(self, domain_space, range_space, linear=True)
+
+        self.displacement = displacement
+
+    def _call(self, template):
+        """Implement ``self(template)```.
+
+        Parameters
+        ----------
+        template : `DiscreteLpVector`
+            Given template that is to be deformed by the fixed
+            displacement field.
+        """
+        div_op = odl.Divergence(range=template.space)
+        jacobian_det = 1 - div_op(self.displacement)
```

Comment:
My point is that while both are not accurate for large displacements, `exp(-div_op(self.displacement))` has significantly better properties in that regime. 

First, it always gives valid values. For example if `div_op(self.displacement) = 2` we get

` exp(-div_op(self.displacement)) = 0.135`
`1 - div_op(self.displacement) = -1`

The first value is always valid, while the second one isnt.

Further, the `exp` way has a proper inverse, while this way does not. For example, if we compose two deformations with `div_op(self.displacement) = +/- 0.5` with this method we get

`jacobian_det = (1 - 0.5) * ( 1 + 0.5) = 0.75`

which is too small, but with the `exp` based method, we get the correct value:

`jacobian_det = exp(- 0.5) * exp(0.5) = 1`